src/ram/transform/MakeIndex.cpp

```diff
@@ -87,7 +87,7 @@ ExpressionPair MakeIndexTransformer::getSignedExpressionPair(
             if (lhs->getTupleId() == identifier && rla->getLevel(rhs) < identifier) {
                 element = lhs->getElement();
                 std::vector<std::unique_ptr<RamExpression>> expressions;
-                expressions.push_back(std::unique_ptr<RamExpression>(rhs->clone()));
+                expressions.push_back(std::unique_ptr<RamExpression>(clone(rhs)));
```

Comment:
clone returns a unique_ptr so you can just do 
```suggestion
                expressions.push_back(clone(rhs));
```